pkg/api/types.go

```diff
@@ -1004,26 +1008,58 @@ type SourceStepConfiguration struct {
 	ClonerefsPath string `json:"clonerefs_path"`
 }
 
+// OperatorStepConfiguration describes the locations of operator bundle information,
+// bundle build dockerfiles, and images the operator(s) depends on that must
+// be substituted to run in a CI test cluster
+type OperatorStepConfiguration struct {
+	// DockerfilePath describes the path(s) to the dockerfile(s) that builds the operator bundle
+	DockerfilePath []string `json:"dockerfile_path,omitempty"`
+
+	// Manifests describes the path(s) of the operator manifests that the bundle(s) requires
+	Manifests []string `json:"manifests,omitempty"`
+
+	// Substitutions describes the pullspecs in the operator manifests that must be subsituted
+	// with the pull specs of the images in the CI registry
+	Substitutions []PullSpecSubstitution `json:"substitutions,omitempty"`
+}
+
+// IndexGeneratorStepConfiguration describes a step that creates an index database and
+// Dockerfile to build an operator index that uses the generated database based on
+// bundle names provided in OperatorIndex
+type IndexGeneratorStepConfiguration struct {
+	To PipelineImageStreamTagReference `json:"to,omitempty"`
+
+	// OperatorIndex is a list of the names of the bundle images that the
+	// index will contain in its database.
+	OperatorIndex []string `json:"operator_index,omitempty"`
+}
+
+// IndexImageGeneratorName is the name of the index image generator built by ci-operator
+const IndexImageGeneratorName = "ci-index-gen"
+
+// IndexImageName is the name of the index image built by ci-operator
+const IndexImageName = "ci-index"
+
 // BundleSourceStepConfiguration describes a step that performs a set of
 // substitutions on operator manifests in the `src` image so that the
 // pullspecs in the operator manifests point to images inside the CI registry.
 // It is intended to be used as the source image for bundle image builds.
 type BundleSourceStepConfiguration struct {
-	To PipelineImageStreamTagReference `json:"to,omitempty"`
-
-	// ContextDir is the directory in the project
-	// from which this build should be run.
-	ContextDir string `json:"context_dir,omitempty"`
-
-	// OperatorManifests is the subdir of context_dir where optional
+	// Manifests is the subdir of context_dir where optional
 	// operator manifests are stored for operator bundle images.
-	OperatorManifests string `json:"operator_manifests,omitempty"`
+	Manifests []string `json:"operator_manifests,omitempty"`
```

Comment:
Why is this a slice?

Reply:
Also don't we still need `to` and `context_dir` on this config? What if someone adds one via a `raw_step`?